### PR TITLE
feature(new integration): Attentive Tag Cloud mode integration Onboarding

### PR DESCRIPTION
**Pull request by ujjwal-ab**
## Description of the change

> Attentive Tag Cloud mode integration Onboarding

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/attentive_tag/transform.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  defaultPostRequestConfig,
+  removeUndefinedAndNullValues,
+  getIntegrationsObj,
+  CustomError
+} = require("../../util");
+
+const identifyResponseBuilder = (message, { Config }) => {
+  const { apiKey, signUpSourceId } = Config;
+  let endpoint; 
+  const integrationsObj = getIntegrationsObj(message, "attentive_tag");
```
Comment (shrouti1507) [inline v0/destinations/attentive_tag/transform.js:17]: Add canonical names also in constants/destinationCanonicalNames.js

File: v0/destinations/attentive_tag/transform.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    };
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+  }
+  return response;
+};
+const trackResponseBuilder = (message, { Config }) => {
+  const { apiKey } = Config;
+
+  let event = get(message, "event");
```
Comment (shrouti1507) [inline v0/destinations/attentive_tag/transform.js:47]: Throw error is event key is not present.

File: v0/destinations/attentive_tag/transform.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  if (payload) {
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}${endpoint}`;
+    response.method = defaultPostRequestConfig.requestMethod;
+    response.headers = {
+      Authorization: `Bearer ${apiKey}`,
+      "Content-Type": "application/json"
+    };
+    response.body.JSON = removeUndefinedAndNullValues(payload);
+    return response;
```
Comment (shrouti1507) [inline v0/destinations/attentive_tag/transform.js:97]: line 88-97 is repetitive with line no 32-40. Move it to a separate function and write once.

File: v0/destinations/attentive_tag/transform.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+const identifyResponseBuilder = (message, { Config }) => {
+  const { apiKey, signUpSourceId } = Config;
+  let endpoint; 
+  const integrationsObj = getIntegrationsObj(message, "attentive_tag");
+  let payload;
+  if (integrationsObj) {
+    if (integrationsObj.signUpSourceId) {
+      signUpSourceId = integrationsObj.signUpSourceId;
+    }
+    if (integrationsObj?.identifyOperation.toLowerCase() === "unsubscribe") {
```
Comment (shrouti1507) [inline v0/destinations/attentive_tag/transform.js:23]: add the axios call here to check if the user is present in the database at all

File: v0/destinations/attentive_tag/transform.js (hunk 2 above)
Comment (shrouti1507) [reply v0/destinations/attentive_tag/transform.js:64]: address this

File: v0/destinations/attentive_tag/transform.js (hunk 1 above)
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/transform.js:25]: Added

File: v0/destinations/attentive_tag/transform.js (hunk 4 above)
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/transform.js]: Added

File: v0/destinations/attentive_tag/util.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  } catch (err) {
+    // check if exists err.response && err.response.status else 500
+    if (err.response && err.response.status) {
+      throw new CustomError(err.response.statusText, err.response.status);
+    }
+    throw new CustomError("[Attentive_Tag] :: Failed to make request", 500);
+  }
+}
+function getExternalIdentifiersMapping(message) {
+  const externalIdentifiers = ["clientUserId", "shopifyId", "klaviyoId"];
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/util.js:32]: Done

File: v0/destinations/attentive_tag/transform.js (hunk 3 above)
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/transform.js:133]: Done

File: v0/destinations/attentive_tag/util.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+    const price = [];
+    const props = {};
+    const pricing = {};
+    const properties = get(message, "properties");
+    props.productId = properties.productId;
+    props.productVariantId = properties.variant;
+    props.name = properties.name;
+    props.productImage = properties.image_url;
+    props.productUrl = properties.url;
+    props.quantity = properties.quantity;
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/util.js]: Updated

File: v0/destinations/attentive_tag/transform.js [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = await identifyResponseBuilder(message, destination);
+      break;
+    case EventType.TRACK:
+      response = trackResponseBuilder(message, destination);
+      break;
+    default:
+      throw new Error("Message type not supported");
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/transform.js]: done

File: v0/destinations/attentive_tag/data/attentiveUnsubscribeConfig.json [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+    "sourceKeys": "phone",
+    "sourceFromGenericMap": true,
+    "required": false
+  },
+  {
+    "destKey": "user.email",
+    "sourceKeys": "email",
+    "sourceFromGenericMap": true,
+    "required": false
+  }
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/data/attentiveUnsubscribeConfig.json:13]: done

File: v0/destinations/attentive_tag/transform.js [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+      payload.subscriptions = integrationsObj?.subscriptions;
+      payload.notification = integrationsObj?.language;
+    }
+    if (integrationsObj?.identifyOperation?.toLowerCase() === "subscribe") {
+      endpoint = "/subscriptions";
+      payload = constructPayload(
+        message,
+        mappingConfig[ConfigCategory.SUBSCRIBE.name]
+      );
+      payload.signUpSourceId = signUpSourceId;
```
[the remark this answers is not in the document]
Comment (ujjwal-ab) [reply v0/destinations/attentive_tag/transform.js:64]: done

**Review by shrouti1507 — APPROVED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**